### ISSUE TITLE
add_, addcmul_, addcdiv_ deprecated

### ISSUE THREAD

**Issue by jecampagne** [pytorch/pytorch#32861]
## Do someone else experiences many messages due to deprecated functions?

Using `torch 1.5.0.dev20200129` to track a bug, I experience these Warning. Do the library will change its API???

/opt/conda/conda-bld/pytorch_1580285301452/work/torch/csrc/autograd/generated/python_variable_methods.cpp:1334: UserWarning: This overload of add_ is deprecated:
add_(Number alpha, Tensor other)
Consider using one of the following signatures instead:
add_(Tensor other, Number alpha)
/opt/conda/conda-bld/pytorch_1580285301452/work/torch/csrc/autograd/generated/python_variable_methods.cpp:1550: UserWarning: This overload of addcmul_ is deprecated:
addcmul_(Number value, Tensor tensor1, Tensor tensor2)
Consider using one of the following signatures instead:
addcmul_(Tensor tensor1, Tensor tensor2, Number value)
/opt/conda/conda-bld/pytorch_1580285301452/work/torch/csrc/autograd/generated/python_variable_methods.cpp:1480: UserWarning: This overload of addcdiv_ is deprecated:
addcdiv_(Number value, Tensor tensor1, Tensor tensor2)
Consider using one of the following signatures instead:
addcdiv_(Tensor tensor1, Tensor tensor2, Number value)

**Comment by hsl89** [comment]
Yes, I have encountered the same warning. This happened after I build my custom Dataset object for loading images (cifar10). 
I saved cifar10 images as png images in `train/` and `test/` dirs. Here is my custom Dataset 

`
from torch.utils.data import Dataset
import os
import PIL.Image as Image

class ImageDataCLS(Dataset):
    '''Image dataset object for classification'''
    def __init__(self, root, train=True, transform=None):
        '''
        root(str): root of data directory
        train (bool): use as train set
        transform(callable) transform applied to imgs
        '''
        self.transform=transform

        if train:
            imgdir=os.path.join(root, 'train')
            labels='train_labels.txt'
        else:
            imgdir=os.path.join(root, 'test')
            labels='test_labels.txt'

        # load images path and labels
        self.fpath=[]
        for im in os.listdir(imgdir):
            self.fpath.append(
                os.path.join(root, imgdir, im))
            
        self.target=[]
        with open(os.path.join(root, labels), 'r') as f:
            for ln in f.readlines():
                t = ln.split(',')[-1].strip()
                self.target.append(int(t))

    def __len___(self):
        return len(self.target)
    
    def __getitem__(self, idx):
        img = Image.open(self.fpath[idx])
        t = self.target[idx]

        if self.transform:
            img = self.transform(img)
        return img, t


`

**Comment by zasdfgbnm** [comment]
cc: @xwang233 Does your fix cover this bug?

**Comment by xwang233** [comment]
Most of these calls with deprecated signatures were fixed in #33428. They were mainly in optimizer and test code. Thanks for checking, and please let me know if there is anything missed.

**Comment by zasdfgbnm** [comment]
@jecampagne These APIs were deprecated long time ago, but just recently PyTorch start to issue warnings to them. Even inside PyTorch, these APIs were used, which is fixed at https://github.com/pytorch/pytorch/pull/33428 by @xwang233. Could you try the latest nightly build to see if you can still see these warnings? These warnings are most likely coming from optimizers.

**Comment by nihiluis** [comment]
```
/pytorch/torch/csrc/utils/python_arg_parser.cpp:756: UserWarning: This overload of add_ is deprecated:
	add_(Number alpha, Tensor other)
Consider using one of the following signatures instead:
	add_(Tensor other, *, Number alpha)
```

Not sure if this is relevant here

**Comment by xwang233** [comment]
Hi @nihiluis , thanks for the report! Yes, this is relevant. Most pytorch code using `add` should already be fixed in this repo except some tests. Can you provide your use case? Are you running code from a third party library? If so, they may have some python code that is not updated so far.

**Comment by pommedeterresautee** [comment]
Training with 2.8 huggingface transformers library produces this warning (standard training script from the lib).

**Comment by nihiluis** [comment]
Thank you boss.. yes it is correct.. i use this lib.

**Comment by brando90** [comment]
can one give an example of how to fix this? I tried and I still can't:

```
                # exp_avg_sq.mul_(beta2).addcmul_(1 - beta2, grad, grad)
                one_minus_beta2: torch.Tensor = torch.tensor(1 - beta2, device=grad.device)
                exp_avg_sq.mul_(beta2).addcmul_(one_minus_beta2, grad, grad)
                exp_avg.mul_(beta1).add_(1 - beta1, grad)
```
err
```
/Users/miranda9/RAdam/radam/radam.py:60: UserWarning: This overload of addcmul_ is deprecated:
	addcmul_(Number value, Tensor tensor1, Tensor tensor2)
Consider using one of the following signatures instead:
	addcmul_(Tensor tensor1, Tensor tensor2, *, Number value) (Triggered internally at  /Users/distiller/project/conda/conda-bld/pytorch_1623459065530/work/torch/csrc/utils/python_arg_parser.cpp:1025.)
  exp_avg_sq.mul_(beta2).addcmul_(one_minus_beta2, grad, grad)
```

**Comment by brando90** [comment]
related: https://github.com/lessw2020/Ranger-Deep-Learning-Optimizer/issues/40

**Comment by brando90** [comment]
https://discuss.pytorch.org/t/userwarning-this-overload-of-add-addcmul-addcdiv-is-deprecated-errors-while-implementing-sharedadam/99802/2

**Comment by mruberry** [comment]
I think this behavior is expected. You just need to update the calls to pass alpha as a kwarg at the end of the call to add:

```
torch.add(..., alpha=...)
```

And do the same with addcmul and addcdiv but with the keyword "value."
